src/redux/books/books.js

```diff
@@ -1,56 +1,79 @@
-import { createSlice, nanoid } from '@reduxjs/toolkit';
+import BookService from '../../services/BooksService';
 
-const initialState = {
-  books: [
-    {
-      id: 0,
-      title: 'The Hunger Games',
-      categoryId: 1,
-    },
-    {
-      id: 1,
-      title: 'Capital in the Twenty-First Century',
-      categoryId: 2,
-    },
-  ],
+const GET_ALL_BOOKS = 'books/GET_ALL_BOOKS';
+const ADD_BOOK = 'books/ADD_BOOK';
+const REMOVE_BOOK = 'books/REMOVE_BOOK';
+
+const initialState = [];
+
+export const getAllBooks = () => async (dispatch) => {
+  try {
+    const res = await BookService.getAllBooks();
+    dispatch({
+      type: GET_ALL_BOOKS,
+      payload: res.data,
+    });
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+// eslint-disable-next-line camelcase
+export const addBook = (item_id, title, author, category) => async (dispatch) => {
+  try {
+    const res = await BookService.addBook({
+      // eslint-disable-next-line camelcase
+      item_id, title, author, category,
+    });
+    dispatch({
+      type: ADD_BOOK,
+      payload: res.data,
+    });
+    return Promise.resolve(res.data);
+  } catch (err) {
+    return Promise.reject(err);
+  }
 };
```

Comment:
- [ ] After I enter the details of a book and click on `Add Book` button, nothing happened and the book was not added to the Home page. I have checked the console and I found the following error:

<img width="355" alt="image" src="https://user-images.githubusercontent.com/77190203/187022390-094a51a8-f930-42bf-97ef-a39dc704759e.png">

As the error shows, `unwrap` is not a function. Please try to fix the error at the `saveBook` method in the  `AddBookForm` component, and ensure that a new book is added successfully via the API endpoint. The added book should be shown immediately on the Home page after adding it.